pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,164 @@
+			imageRepoName := cmdutil.GetFlagString(cmd, "imagerepo")
+			if imageRepoName == "" {
+				usageError(cmd, "Must pass an image repository name with the --imagerepo (-i) flag")
+			}
+
+			c, _, err := f.Clients(cmd)
+			checkErr(err)
+
+			namespace, err := f.DefaultNamespace(cmd)
+			checkErr(err)
+
+			// Validate the specified image repository
+			if !isValid(imageRepoName, c, namespace) {
+				checkErr(fmt.Errorf("invalid image repository: %s", imageRepoName))
+			}
+
+			// Get all build configurations and look for those
+			// that reference the specified image repository
+			buildConfigList, err := c.BuildConfigs(namespace).List(labels.Everything(), labels.Everything())
+			checkErr(err)
+
+			root := findRepoDeps(imageRepoName, buildConfigList.Items)
+			fmt.Println(root)
+			// imgRoot := repoToImage(c, namespace, imageRepoName, root)
+		},
+	}
+
+	cmd.Flags().StringP("imagerepo", "i", "", "Image repository for which to output build dependencies")
+	return cmd
+}
+
+// isValid checks if the given image repository is valid
+func isValid(name string, c *client.Client, namespace string) bool {
+	// TODO: Validate image repository via buildConfigs?
+	repoList, err := c.ImageRepositories(namespace).List(labels.Everything(), labels.Everything())
+	checkErr(err)
+
+	for _, repo := range repoList.Items {
+		if name == repo.Name {
+			if errs := val.ValidateImageRepository(&repo); len(errs) == 0 {
+				return true
+			}
+			break
+		}
+	}
+
+	return false
+}
+
+// findRepoDeps accepts an image repository name and a list of build
+// configurations and returns a dependency tree of the specified
+// image repository along with its tags
+func findRepoDeps(from string, buildConfigList []buildapi.BuildConfig) *node {
+	root := &node{
+		name: from,
+	}
+
+	// Search all build configurations in order to find the image
+	// repositories depending on the specified image repository
+	for _, cfg := range buildConfigList {
+		for _, trigger := range cfg.Triggers {
+			if from == trigger.ImageChange.From.Name {
+				// Search recursively all children for their dependency trees
+				// TODO: Check for both outputs ? (To & Tag or DockerImageReference)
+				child := findRepoDeps(cfg.Parameters.Output.To.Name, buildConfigList)
+				child.tags = append(child.tags, cfg.Parameters.Output.Tag)
+				root.children = append(root.children, child)
+			}
+		}
+	}
+
+	return root
+}
+
+// repoToImage accepts an image repository tree and returns its equivalent
+// image tree. The root node keeps on representing the image repository
+// specified with the --imagerepo (-i) flag though
+func repoToImage(c *client.Client, namespace string, nodeName string, root *node) *node {
+	if root == nil {
+		return nil
+	}
+	nd := &node{
+		name: nodeName,
+	}
+
+	for _, child := range root.children {
+		// Search through tags
+		for _, tag := range child.tags {
+			// TODO: These images probably do not exist since we haven't run any buildConfigs
```

Comment:
Right- we should output the other value, and maybe in a later pull try to infer overlap / reductions


> On Mar 10, 2015, at 11:51 AM, Cesar Wong <notifications@github.com> wrote:
> 
> In pkg/cmd/experimental/buildchain/buildchain.go:
> 
> > +
> > +// repoToImage accepts an image repository tree and returns its equivalent
> > +// image tree. The root node keeps on representing the image repository
> > +// specified with the --imagerepo (-i) flag though
> > +func repoToImage(c *client.Client, namespace string, nodeName string, root *node) *node {
> > +	if root == nil {
> > +		return nil
> > +	}
> > +	nd := &node{
> > +		name: nodeName,
> > +	}
> > +
> > +	for _, child := range root.children {
> > +		// Search through tags
> > +		for _, tag := range child.tags {
> > +			// TODO: These images probably do not exist since we haven't run any buildConfigs
> Correct... which is why I think we should only be looking at the repo relationships (to)
> 
> —
> Reply to this email directly or view it on GitHub.
>